internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -199,3 +221,187 @@ func getDecodedSpecData(data strfmt.Base64) (*dockerConfigJSON, error) {
 
 	return dockerConfigJSON, nil
 }
+
+func constructFullname(d *schema.ResourceData) (fullname *secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretFullName) {
+	fullname = &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretFullName{}
+
+	fullname.ClusterName, _ = d.Get(scope.ClusterNameKey).(string)
+
+	fullname.ManagementClusterName, _ = d.Get(scope.ManagementClusterNameKey).(string)
+
+	fullname.Name, _ = d.Get(NameKey).(string)
+
+	fullname.NamespaceName, _ = d.Get(NamespaceNameKey).(string)
+
+	fullname.OrgID, _ = d.Get(OrgIDKey).(string)
+
+	fullname.ProvisionerName, _ = d.Get(scope.ProvisionerNameKey).(string)
+
+	return fullname
+}
+
+func constructFullnameSecetExport(d *schema.ResourceData) (fullname *secretexportmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretexportFullName) {
+	fullname = &secretexportmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretexportFullName{}
+
+	secretFullname := constructFullname(d)
+
+	fullname.ClusterName = secretFullname.ClusterName
+
+	fullname.ManagementClusterName = secretFullname.ManagementClusterName
+
+	fullname.Name = secretFullname.Name
+
+	fullname.NamespaceName = secretFullname.NamespaceName
+
+	fullname.OrgID = secretFullname.OrgID
+
+	fullname.ProvisionerName = secretFullname.ProvisionerName
+
+	return fullname
+}
+
+func resourceSecretCreate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	secretRequest := &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecretRequest{
+		Secret: &secretmodel.VmwareTanzuManageV1alpha1ClusterNamespaceSecret{
+			FullName: constructFullname(d),
+			Meta:     common.ConstructMeta(d),
+			Spec:     constructSpec(d),
+		},
+	}
+
+	secretResponse, err := config.TMCConnection.SecretResourceService.SecretResourceServiceCreate(secretRequest)
+
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "unable to create Tanzu Mission Control secret entry, name : %s", NameKey))
+	}
+
+	d.SetId(secretResponse.Secret.Meta.UID)
+
+	export := d.Get(ExportKey).(bool)
```

Comment:
this can be used in the if condition itself

Reply:
Updated PTAL